### PR TITLE
fix(treasury-logger): versioned state, do not allow zero amounts and self-deposits

### PR DESCRIPTION
**Pull request by mitinarseny**
## Summary by CodeRabbit

* **Bug Fixes**
  * Reject self-deposit transfers and validate amounts to disallow zero transfers.
  * Ensure emitted deposit events contain accurate token/sender/amount details.

* **Tests**
  * Expanded parameterized test suite covering edge cases, including self-deposits and zero-amount failures.

[Review Change Stack](https://app.coderabbit.ai/change-stack/near/intents/pull/273?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by aleksuss — APPROVED** [review]

File: contracts/treasury-logger/src/lib.rs [hunk 1]
```diff
@@ -45,11 +51,15 @@ impl MultiTokenReceiver for Contract {
...
-            token: env::predecessor_account_id().into(),
+            token: token.into(),
             sender_id: sender_id.into(),
             previous_owner_ids: previous_owner_ids.iter().map(Into::into).collect(),
             token_ids: token_ids.iter().map(Into::into).collect(),
-            amounts: amounts.iter().map(|a| a.0).collect(),
+            amounts: amounts
+                .iter()
+                .map(|a| a.0)
+                .inspect(|a| require!(*a > 0, "zero amount"))
```
Comment (pityjllk) [inline contracts/treasury-logger/src/lib.rs:72]: nit: inspect primerly is ment to be used for debbugin/logging...

File: contracts/treasury-logger/src/lib.rs [hunk 2]
```diff
@@ -103,12 +176,12 @@ mod tests {
                 "version": "1.0.0",
                 "event": "mt_deposit",
                 "data": {
-                    "token": "intents.near",
-                    "sender_id": "alice.near",
-                    "previous_owner_ids": ["alice.near"],
-                    "token_ids": ["nep141:wrap.near"],
-                    "amounts": ["100"],
-                    "msg": "test",
+                    "token": token,
```
Comment (pityjllk) [inline contracts/treasury-logger/src/lib.rs]: how about `.to_nep297_event().to_event_log()`  to compare logs?

File: contracts/treasury-logger/src/lib.rs [hunk 3]
```diff
@@ -77,22 +87,85 @@ mod tests {
...
+        [100],
+        "test"
+    )]
+    #[should_panic = "zero amount"]
+    #[case::zero_amount_panics(
+        "intents.near",
+        "alice.near",
+        ["alice.near"],
+        ["nep141:wrap.near"],
+        [0],
```
Comment (pityjllk) [inline contracts/treasury-logger/src/lib.rs]: sould be nice to have test that fails where only  one of the send values is 0

**Review by pityjllk — CHANGES_REQUESTED** [review]
